03_letters/lexual/freq_test.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+)
+
+func TestLetters(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected map[rune]int
+	}{
+		{
+			"abcbcc",
+			map[rune]int{
+				'a': 1,
+				'b': 2,
+				'c': 3,
+			},
+		},
+		{
+			"",
+			map[rune]int{},
+		},
+	}
+	for _, test := range tests {
+		actual := letters(test.input)
+		if len(actual) != len(test.expected) {
+			t.Errorf("len of actual and expected differ")
+		}
+		for k, v := range test.expected {
+			if v != actual[k] {
+				t.Errorf("Unexpected result")
+				t.Errorf("\nActual: %v\nExpected: %v", actual, test.expected)
+				for k, v := range actual {
```

Comment:
I really don't think the looping over the `actual` and `expected` is beneficial here. It looks like you are doing this just for the purpose of printing, but you have already printed both the actual and expected above. This just introduces two iterations which are not required. Please remove this.

I would recommend running these test and providing an input which will make them fail. That way you will see what is printed and you will be able to determine if this is overkill. You will see only one `t.Errorf` should be required

Reply:
its because printing the map, shows the keys as integers instead of letters.

i.e. 

```Actual: map[97:1 98:2 99:3]
Expected: map[97:1 98:22 99:3]```

Isn't very useful for debugging, because I need to determine in my head that 97 corresponds to 'a'.

Full output gives this:

```freq_test.go:35: 
    Actual: map[97:1 98:2 99:3]
    Expected: map[97:1 98:22 99:3]
freq_test.go:37: Actual a = 1
freq_test.go:37: Actual b = 2
freq_test.go:37: Actual c = 3
freq_test.go:40: Expect c = 3
freq_test.go:40: Expect a = 1
freq_test.go:40: Expect b = 22
```